Firestore/Source/API/FIRDocumentReference.mm

```diff
@@ -200,6 +210,20 @@ - (FSTListenOptions *)internalOptionsForIncludeMetadataChanges:(BOOL)includeMeta
                                                  waitForSyncWhenOnline:NO];
 }
 
+- (HandleMaybe<DocumentSnapshot>)wrapDocumentSnapshotBlock:(FIRDocumentSnapshotBlock)block {
+  FIRFirestore *firestore = _firestore;
+  return [block, firestore](StatusOr<DocumentSnapshot> maybe_snapshot) {
```

Comment:
This will probably have to move somewhere else later on, this basically translates a `std::function` taking a `StatusOr` into a block taking two parameters, one of which is an `NSError`.